pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -181,13 +182,19 @@ func handleError(message string, listOfErrors ...error) error {
 }
 
 // ValidateBuildRegistration verifies that a referenced Build is properly register
-func (r *ReconcileBuildRun) ValidateBuildRegistration(ctx context.Context, build *buildv1alpha1.Build, buildRun *buildv1alpha1.BuildRun) error {
+func (r *ReconcileBuildRun) ValidateBuildRegistration(ctx context.Context, build *buildv1alpha1.Build, buildRun *buildv1alpha1.BuildRun) (*reconcile.Result, error) {
+	if build.Status.Registered == "" {
+		ctxlog.Info(ctx, "Build is not yet validated. Checking again in five seconds.", namespace, buildRun.Namespace, name, buildRun.Name, "build", build.Name)
+		return &reconcile.Result{
+			RequeueAfter: 5 * time.Second,
```

Comment:
I took a look on this. Actually if you return an error, k8s is smart enough to increase the times between reconciles. I think after reconciling 2-3 times, then the reconciles start happening 2times the initial delta. Meaning if this error will end up in 10 reconciles, between the reconcile 9 the 10 there will be more than one minute.